django_sharding_library/decorators.py

```diff
@@ -24,12 +30,30 @@ def configure(cls):
                 )
             setattr(cls, 'django_sharding__database', database)
 
+        postgres_shard_id_fields = list(filter(lambda field: issubclass(type(field), PostgresShardGeneratedIDField), cls._meta.fields))
+        if postgres_shard_id_fields:
+            # Make sure we are only using postgres
+            if database:
+                if settings.DATABASES[database]['ENGINE'] not in Backends.POSTGRES:
+                    raise ShardedModelInitializationException('You cannot use a PostgresShardGeneratedIDField on a '
+                                                              'non-Postgres database.')
+
+            if shard_group:
+                for db_key in settings.DATABASES.keys():
+                    if settings.DATABASES[db_key]['ENGINE'] not in Backends.POSTGRES:
+                        raise ShardedModelInitializationException('You cannot use a PostgresShardGeneratedIDField on a '
+                                                                  'non-Postgres database.')
+
+            register_migration_signal_for_model_receiver(apps.get_app_config(cls._meta.app_label),
```

Comment:
I spent more time reading that code for dependencies and it isn't worth trying to hack into it, given they don't expose any good hooks for us to use.